pinecone/client.go

```diff
@@ -408,3 +412,40 @@ func minOne(x int32) int32 {
 	}
 	return x
 }
+
+func buildClientOptions(in NewClientParams) ([]control.ClientOption, error) {
+	clientOptions := []control.ClientOption{}
+	hasAuthorizationHeader := false
+	hasApiKey := in.ApiKey != ""
+
+	userAgentProvider := provider.NewHeaderProvider("User-Agent", useragent.BuildUserAgent(in.SourceTag))
+	clientOptions = append(clientOptions, control.WithRequestEditorFn(userAgentProvider.Intercept))
+
+	for key, value := range in.Headers {
+		headerProvider := provider.NewHeaderProvider(key, value)
+
+		if strings.Contains(key, "Authorization") {
+			hasAuthorizationHeader = true
+		}
```

Comment:
http headers are typically case in-sensitive so we probably want to relax this check

Reply:
Updated to use `strings.ToLower(key), "authorization")`, thanks!